e2e/credentials-server.e2e.spec.ts

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/**
+ * This file includes end-to-end testing for the HTTP control interface
+ */
+
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import { spawnSync } from 'child_process';
+
+import { expect, test } from '@playwright/test';
+import { BrowserContext, ElectronApplication, Page, _electron } from 'playwright';
+
+import fetch from 'node-fetch';
+import { createDefaultSettings, playwrightReportAssets } from './utils/TestUtils';
+import paths from '@/utils/paths';
+import { ServerState } from '@/main/commandServer/httpCommandServer';
+import { spawnFile } from '@/utils/childProcess';
+import { findHomeDir } from '@/config/findHomeDir';
+
+function haveDockerCredentialAssistant(): boolean {
+  // Not using the code from `httpCredentialServer.ts` because we can't use async code at top-level here.
+  const dockerConfigPath = path.join(findHomeDir() ?? '', '.docker', 'config.json');
+
+  try {
+    const contents = JSON.parse(fs.readFileSync(dockerConfigPath).toString());
+    const credStore = contents.credsStore;
+
+    if (!credStore) {
+      return false;
+    }
+    const result = spawnSync(`docker-credential-${ credStore }`, { input: 'list', stdio: 'pipe' });
+
+    return !result.error;
+  } catch (err: any) {
+    return false;
+  }
+}
+
+const testWithCreds = haveDockerCredentialAssistant() ? test : test.skip;
+
+test.describe('Credentials server', () => {
+  let electronApp: ElectronApplication;
+  let context: BrowserContext;
+  let serverState: ServerState;
+  let authString: string;
+  let page: Page;
+  const appPath = path.join(__dirname, '../');
+  const curl = os.platform() === 'win32' ? path.join(process.env['SYSTEM_ROOT'] ?? 'c:\\windows', 'system32', 'curl.exe') : 'curl';
+
+  async function doRequest(path: string, body = '') {
+    const args = [
+      '-s',
+      '-H',
+      `Authorization: Basic ${ authString }`,
```

Comment:
The `--header` should be on the same line as its value (or format it as `--header=…`).

Reply:
Yes, but using `--user` outside the header.